aten/src/ATen/NumericUtils.h

```diff
@@ -62,6 +64,18 @@ inline C10_HOST_DEVICE bool _isnan(at::BFloat16 val) {
   return at::_isnan(static_cast<float>(val));
 }
 
+template <typename T,
+         typename std::enable_if<std::is_same<T, at::Float8_e5m2>::value, int>::type = 0>
+inline C10_HOST_DEVICE bool _isnan(T val) {
+  return at::_isnan(static_cast<float>(val));
```

Comment:
We probably will need to replace this with a primitive at some point as this is highly unlikely to be optimal.

Reply:
Changed to helper method
```
inline C10_HOST_DEVICE bool Float8_e4m3fn::isnan() const {
  return (x & 0b01111111) == 0b01111111;
}
```